cmd/skaffold/app/cmd/cmd.go

```diff
@@ -40,12 +40,12 @@ var (
 )
 
 var rootCmd = &cobra.Command{
-	Use:   "skaffold",
-	Short: "A tool that facilitates continuous development for Kubernetes applications.",
+	Use:          "skaffold",
+	Short:        "A tool that facilitates continuous development for Kubernetes applications.",
+	SilenceUsage: true,
```

Comment:
If you move this line to `PersistentPreRunE` function, it will silent usage only after args are successfully checked.

Reply:
or use https://github.com/GoogleCloudPlatform/runtimes-common/tree/master/ctc_lib
Where we silenceUsage Errors in initializers.